### PR TITLE
Graph collector deadline header v2

### PR DESCRIPTION
**Pull request by salegria8**
### Description
The deadline header will be ignored in older versions of 128T so a new config knob doesn't need to be exposed.

### Testing
Unit

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Review by gregschrock — APPROVED** [review]
I do like this. I wonder if we'll get into situations where we'll want a config knob around the difference between the timeouts.

File: plugins/inputs/t128_graphql/t128_graphql.go [hunk 1]
```diff
@@ -240,10 +225,9 @@ func (plugin *T128GraphQL) createRequest() (*http.Request, error) {
 
 	request.Header.Add("Content-Type", "application/json")
 
-	if plugin.Deadline.Duration != DefaultDeadline {
-		deadline := int(plugin.Deadline.Duration.Truncate(time.Second))
-		request.Header.Add("deadline", fmt.Sprintf("%d", deadline))
-	}
+	// ignored in older versions of 128T
+	deadline := int(plugin.Timeout.Duration.Seconds() - timeoutDeadlineDiff.Seconds())
```
Comment (BenMatase) [inline plugins/inputs/t128_graphql/t128_graphql.go:229]: If someone put 1s or 0s for timeout? Does graph handle `0` or `-1` deadline?

**Review by BenMatase — APPROVED** [review]
I like it

File: plugins/inputs/t128_graphql/t128_graphql.go (hunk 1 above)
Comment (salegria8) [reply plugins/inputs/t128_graphql/t128_graphql.go:229]: good point. i'll ask Dillon and might need to add a minimum timeout

Comment (salegria8) [reply plugins/inputs/t128_graphql/t128_graphql.go:229]: Dillon's going to add a guard to ignore negative and 0sec deadlines. So if a user configures `timeout = "1s"` then `deadline = 0s` which will be treated by web as an infinite deadline. But we're still covered by the 1s timeout

**Comment by agrawalkaushik** [comment]
> I do like this. I wonder if we'll get into situations where we'll want a config knob around the difference between the timeouts.

The timeout can be adjusted in those situations. We should not add knobs for internal implementation details like these.

**Pull request merged**